src/module_2/EDA_1.md

```diff
@@ -0,0 +1,3314 @@
+
+# Calcular el número de ítems por pedido
+orders['num_items'] = orders['ordered_items'].apply(len)
+
+# Promedio de ítems por pedido
+avg_items_per_order = orders['num_items'].mean()
+print(f"Promedio de ítems por pedido: {avg_items_per_order:.2f}")
+import matplotlib.pyplot as plt
+import seaborn as sns
+
+# Histograma de la distribución de ítems por pedido
+sns.histplot(orders['ordered_items'], bins=20)
+plt.title('Distribución de Ítems por Pedido')
+plt.xlabel('Número de Ítems')
+plt.ylabel('Frecuencia')
+plt.show()
+
+# Asegúrate de que la columna de fecha esté en formato datetime
+orders['order_date'] = pd.to_datetime(orders['order_date'])
+
+# Agrupar por mes y contar las órdenes
+orders_per_month = orders.groupby(orders['order_date'].dt.to_period('M')).size()
+
+# Visualizar la tendencia de órdenes a lo largo del tiempo
+orders_per_month.plot(kind='line', marker='o', figsize=(10,6))
+plt.title('Tendencia de Órdenes a lo Largo del Tiempo')
+plt.xlabel('Fecha')
+plt.ylabel('Número de Órdenes')
+plt.show()
+
+from collections import Counter
+
+# Aplanar la lista de ítems comprados
+all_items = [item for sublist in orders['ordered_items'] for item in sublist]
+
+# Contar la frecuencia de cada ítem
+item_counts = Counter(all_items)
+
+# Convertir a un DataFrame y mostrar los productos más comprados
+item_counts_df = pd.DataFrame(item_counts.items(), columns=['item_id', 'count']).sort_values(by='count', ascending=False)
+
+print("Productos más comprados:")
+print(item_counts_df.head())
+
+# Visualización de los productos más comprados
+top_items = item_counts_df.head(10)
+sns.barplot(x='count', y='item_id', data=top_items)
+plt.title('Top 10 Productos Más Comprados')
+plt.xlabel('Frecuencia')
+plt.ylabel('ID del Producto')
+plt.show()
+
+```
+
+    Número total de órdenes: 8773
+    Promedio de ítems por pedido: 128.00
+
+
+
+    
+![png](EDA_1_files/EDA_1_6_1.png)
+    
+
+
+
+    
+![png](EDA_1_files/EDA_1_6_2.png)
+    
+
+
+    Productos más comprados:
+      item_id  count
+    0       6  71012
+    3       7  70986
+    8       4  70983
+    1       2  70949
+    7       8  70813
+
+
+
+    
+![png](EDA_1_files/EDA_1_6_4.png)
+    
+
+
+
+```python
+
+#RegularsAnalysis
+# Tamaño y forma del archivo
+print(f"Shape: {regulars.shape}")
+print(f"Columns: {regulars.columns}")
+
+# Número de usuarios que tienen ítems regulares
+num_users_with_regulars = regulars['user_id'].nunique()
+print(f"Número de usuarios con ítems regulares: {num_users_with_regulars}")
+
+# Contar la frecuencia de cada producto (variant_id) configurado como regular
+most_common_regular_items = regulars['variant_id'].value_counts().reset_index()
+most_common_regular_items.columns = ['variant_id', 'count']
+
+# Mostrar los productos más comunes como regulares
+print("Items más comunes entre los regulares:")
+print(most_common_regular_items.head())
+
+import seaborn as sns
+import matplotlib.pyplot as plt
+
+# Visualización de los productos más comunes como regulares
```

Comment:
It cannot be that you label an axis "product id" and you see a float like "3.5e13" and you do not reconsider the plot. It is impossible to tell which product is from "3.5e13"

Reply:
In addition a table is much more informative than this graph. In addition, what sense does it make to have vertical bars, with frequency on x axis?